x-pack/plugin/sql/qa/server/src/main/resources/functions.csv-spec

```diff
@@ -347,8 +347,8 @@ SELECT CAST(LOCATE(CASE WHEN languages > 3 THEN 'a' ELSE 'b' END, CASE WHEN TRUN
 
       x:s      |       c:l       
 ---------------+---------------
-false          |80             
-true           |16             
+false          |83             
+true           |13             
```

Comment:
Especially curious about your opinions on this one:

These changes are the result of `start=NULL{f}` argument handling (this is an explicit `start=NULL` vs leaving out the optional parameter).

Previously:
`LOCATE(pattern,input,start=NULL{field})` was handled as `LOCATE(pattern,input,start=1)` or `LOCATE(pattern, input)`, but `LOCATE(pattern,input,start=NULL{literal})` resulted in `NULL`.

Now in both cases we return with 0 to be in-sync with the MySQL behaviour (where `LOCATE` comes from).

MySQL behaviour: `LOCATE(pattern,input,start=NULL) => 0`
MS SQL behaviour: `CHARINDEX(pattern,input,start=NULL) => NULL`

Reply:
Overall I think it would follow the MySQL behaviour since `LOCATE` comes from there.
If we had to change the behaviour only for a literal NULL I would say to go ahead and provide it as a bug fix, but since now we changes how it behaves for a NULL coming from a field, we should really consider. I guess we can always declare it a breaking change?

Reply:
I reverted to the alternative approach and made the `LOCATE(pattern,input,start=NULL{literal})` mimic the current behaviour of the `LOCATE(pattern,input,start=NULL{field})` in this PR. Our `LOCATE` also handles the negative `start` values differently from MySQL. `start=missing/NULL/negative value` all result the same as `start=1` (tries to find the pattern in the whole input) while MySQL considers `start=negative value/NULL` as invalid and returns with `0` (no match). Because `LOCATE` treats `NULL` as the `start` argument specially even in MySQL (not the same as any other argument being `NULL`) I will address these differences in a separate PR.

I am still adding the `breaking change` label to this PR, since the behaviour of the functions change anyways (`NULL` field values will start returning `NULL` values instead of the original input strings).